Mathlib/Topology/Bases.lean

```diff
@@ -124,16 +124,13 @@ member of `s`, then `s` is a topological basis. -/
 theorem isTopologicalBasis_of_isOpen_of_nhds {s : Set (Set α)} (h_open : ∀ u ∈ s, IsOpen u)
```

Comment:
Could you please add `IsTopologicalBasis.of_hasBasis_nhds` version that assumes `h_open` and your first `have`? While the `HasBasis` assumption is formally stronger, we may have a `HasBasis` statement coming from some generic lemma about `Filter`s.

Reply:
I originally added this version with the hope that it would simplify some proofs below, but it didn't so I just removed it. I can add it back if you want.

Reply:
Let me try.

Reply:
I golfed a bit by using `aesop`. Feel free to revert if you don't like it.

Reply:
Ah right, I didn't try aesop. I'll have a closer look but it looks good.